src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
```

Comment:
@AtsushiKan it's not the same. ResolveSimpleAssemblyName first tries to load the assembly with an AssemblyName type and if that fails it tries to load it with the name of the assembly as a string.

At this point we are first trying to resolve it by assemblyname (string) and if that fails by AssemblyName type. So vice versa.